docs/user-guide/templates/pipeline-templates.md

```diff
@@ -0,0 +1,194 @@
+
+* `nodejs/test@latest` - this will use the most recently published version of the template, which may include backwards-incompatible changes, major version changes, etc. The `latest` tag should primarily be used by a template's maintainers and may be unsuitable for production or similarly important builds.
+* `nodejs/test@stable` - this will use a version of the template that its maintainers have designated as sufficiently stable for general usage. It may represent a significant change in capability from older uses of the `stable` tag. Template maintainers should communicate to users when changes to the `stable` tag are not backwards compatible.
+* `nodejs/test@1` - this will use the most recent version of `nodejs/test` that is less than 2.0.0. This is essentially the `latest` tag without crossing a major version boundary.
+* `nodejs/test@1.0` - this will use the most recent version of `nodejs/test` than is less than 1.1.0. This is essentially the `latest` tag without crossing a minor version boundary.
+* `nodejs/test@1.0.4` - this is the most predictable way to specify a pipeline's behavior and is not affected by future changes to the template.
+
+#### Example
+
+For this configuration:
+```yaml
+template: nodejs/test@stable
+shared: 
+  environment:
+    FOO: bar
+```
+
+
+Screwdriver takes the template configuration and plugs it in, so that the `screwdriver.yaml` becomes:
+
+```yaml
+shared:
+  environment:
+    FOO: bar
+jobs:
+    main:
+      image: node:18
+      requires: [~pr, ~commit]
+      steps:
+        - install: npm install
+        - test: npm test
+      secrets:
+            - NPM_TOKEN
+```
+
+## Creating a template
+
+Publishing and running pipeline templates must be done from a Screwdriver pipeline.
+
+### Writing a template yaml
+
+To create a template, create a new repo with a `sd-template.yaml` file. The file should contain a namespace, name, version, description, maintainer email, and a config with jobs. Basic example can be found in our [screwdriver-cd-test/pipe;line-template-example repo](https://github.com/screwdriver-cd-test/pipeline-template-example).
+
+Example `sd-template.yaml`:
+
+```yaml
+namespace: myNamespace
+name: template_name
+version: '1.3'
+description: pipeline template for testing
+maintainer: foo@bar.com
+config:
+    jobs:
+      main:
+        requires: [~pr, ~commit]
+        steps:
+          - install: npm install
+          - test: npm test
+        environment:
+            FOO: bar
+        secrets:
+            - NPM_TOKEN
+```
+
+### Writing a screwdriver yaml for your template repo
+
+#### Validating templates
+
+To validate your template, run the `pipeline-template-validate` script from the `screwdriver-template-main` npm module in your `main` job to validate your template. This means the build image must have NodeJS and NPM properly installed to use it. To publish your template, run the `pipeline-template-publish` script from the same module in a separate job.
```

Comment:
can we provide link to: 
`pipeline-template-validate` script
`screwdriver-template-main` npm module
`pipeline-template-publish` script
or at least provide link to the npm module

Reply:
not sure if we need that, there are commands, anyway I have added a linked to the template-main npm module

Reply:
we should update these to use `npx`. See my comment below.